src/components/DropdownFilterContainer.ts

```diff
@@ -0,0 +1,147 @@
+import * as dijitRegistry from "dijit/registry";
+import * as classNames from "classnames";
+import * as dojoLang from "dojo/_base/lang";
+import * as dojoConnect from "dojo/_base/connect";
+
+import { DropdownFilter, DropdownFilterProps } from "./DropdownFilter";
+import { ValidateConfigs } from "./ValidateConfigs";
+
+export interface DropdownFilterContainerProps {
+    entity: string;
+    mxform: mxui.lib.form._FormBase;
+    targetGridName: string;
+    values: ValueProps[];
+}
+
+export interface ValueProps {
+    caption: string;
+    comparison: comparisonOptions;
+    attribute: string;
+    value: string;
+    constraint: string;
+    filterMethod: filterMethodOptions;
+}
+
+type filterMethodOptions = "equals" | "contains";
+export type comparisonOptions = "static" | "XPath";
+type HybridConstraint = Array<{ attribute: string; operator: string; value: string; path?: string; }>;
+
+export interface ListView extends mxui.widget._WidgetBase {
+    datasource: {
+        xpathConstraints: string;
+    };
+    _datasource: {
+        _constraints: HybridConstraint | string;
+        _entity: string;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface DropdownFilterState {
+    findingWidget: boolean;
+    targetGrid?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class DropdownFilterContainer extends Component<DropdownFilterContainerProps, DropdownFilterState> {
+
+    constructor(props: DropdownFilterContainerProps) {
+        super(props);
+
+        this.state = { findingWidget: true };
+        this.handleChange = this.handleChange.bind(this);
+        dojoConnect.connect(props.mxform, "onNavigation", this, dojoLang.hitch(this, this.initDropdownFilter));
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-dropdown-filter")
+            },
+            createElement(ValidateConfigs, {
+                ...this.props as DropdownFilterContainerProps,
+                filterNode: this.state.targetNode,
+                targetGrid: this.state.targetGrid,
+                targetGridName: this.props.targetGridName,
+                validate: !this.state.findingWidget,
+                values: this.props.values
+            }),
+            this.renderDropdownFilter()
+        );
+    }
+
+    private handleChange(value: string, attribute: string, comparison: comparisonOptions, constraint: string, filterMethod: string) {
+        if (comparison === "static") {
+            this.updateByConstraint(value, attribute, filterMethod);
+        } else {
+            this.updateByXpath(constraint);
+        }
+    }
+
+    private updateByXpath(constraint: string) {
+        this.state.targetGrid.datasource.xpathConstraints = constraint;
```

Comment:
Simply overwriting the constraint is not supporting multiple filters on one list view